File: 2024_01/Exercicios/java/aquecimento_2.java
import java.util.Scanner;

public class aquecimento_2 {
    public static int contarMaiusculo(String str){
        int count = 0;

        if(str.equals("")){
            return 0;
        }

        char primeiroCaractere = str.charAt(0);

        if(primeiroCaractere >= 'A' && primeiroCaractere <= 'Z'){
            count = 1;
        }

        return count + contarMaiusculo(str.substring(1));
    }

    public static void main(String[] args){
        Scanner scanner = new Scanner(System.in);
        String line;

        do {
            line = scanner.nextLine();
            if (!line.equals("FIM")) {
                int resultado = contarMaiusculo(line);
                if(resultado > 0) {
                    System.out.println(resultado);
                }
            }
        } while (!line.equals("FIM"));

        scanner.close();
    }
}

File: 2024_01/Exercicios/C/aquecimento_2.c
#include <stdio.h>
#include <string.h>

int contar_maiusculos(char *str){
    int count = 0;
    
    if (*str == '\0'){
        return 0;
    }

    if (*str >= 'A' && *str <= 'Z'){
        count = 1;
    }

    return count + contar_maiusculos(str + 1);
}

int main(){
    char line[80];

        fgets(line, sizeof(line), stdin);
        line[strcspn(line, "\n")] = '\0';
        if(strcmp(line, "FIM") != 0){
            printf("%d\n", contar_maiusculos(line));
        } else{
            return 0;
        }

 return main();
}